### PR TITLE
Minimally constify `Add`

### PR DESCRIPTION
**Pull request by fee1-dead**
* This PR removes the requirement for `impl const` to have a const stability attribute. cc @RalfJung I believe you mentioned that it would make much more sense to require `const_trait`s to have const stability instead. I agree with that sentiment but I don't think that is _required_ for a small scale experimentation like this PR. https://github.com/rust-lang/project-const-traits/issues/16 should definitely be prioritized in the future, but removing the impl check should be good for now as all callers need `const_trait_impl` enabled for any const impl to work.
* This PR is intentionally minimal as constifying other traits can become more complicated (`PartialEq`, for example, would run into requiring implementing it for `str` as that is used in matches, which runs into the implementation for slice equality which uses specialization)

Per the reasons above, anyone who is interested in making traits `const` in the standard library are **strongly encouraged** to reach out to us on the [Zulip channel](https://rust-lang.zulipchat.com/#narrow/channel/419616-t-compiler.2Fproject-const-traits) before proceeding with the work.

cc @rust-lang/project-const-traits 

I believe there is prior approval from libs that we can experiment, so

r? project-const-traits

### CODE REVIEW

**Comment by RalfJung** [comment]
Agreed regarding const stability attributes. We currently do require them for regular `impl` so before we stabilize anything here we need to clean this up, but for now `const_trait_impl` blocks everything so we don't need per-trait tracking.

> Per the reasons above, anyone who is interested in making traits const in the standard library are strongly encouraged to reach out to us on the [Zulip channel](https://rust-lang.zulipchat.com/#narrow/channel/419616-t-compiler.2Fproject-const-traits) before proceeding with the work.

I also feel like we should repeat what we did last time and have a big "const trait everything" spree until we are confident that the system works and we can stabilize the first parts. We need to do experimentation to gain that confidence, of course, but IMO last time we went too far in using this unstable feature throughout the standard library before it was ready.

File: tests/ui/traits/const-traits/generic-bound.rs
Comment (RalfJung) [inline tests/ui/traits/const-traits/generic-bound.rs:1]: Could this test also call `twice` from const context?

**Review by compiler-errors — APPROVED** [review]
r=me after test tweak

**Pull request merged**